arm9/source/main.c

```diff
@@ -66,19 +72,19 @@ int main(int argc, char *argv[])
 		wait_any_key_poweroff();
 	}
 
-	if (!load_file(FALLBACK_PATH LINUXIMAGE_FILENAME, ZIMAGE_ADDR)) {
-		if (!load_file(LINUXIMAGE_FILENAME, ZIMAGE_ADDR))
-			goto error;
+	if (!load_file(LINUXIMAGE_FILENAME, ZIMAGE_ADDR)) {
+		Debug("Failed to load " LINUXIMAGE_FILENAME);
+		goto error;
 	}
 
-	if (!load_file(FALLBACK_PATH DTB_FILENAME, PARAMS_ADDR)) {
-		if (!load_file(DTB_FILENAME, PARAMS_ADDR))
-			goto error;
+	dtb_filename = is_ktr() ? KTR_DTB_FILENAME : CTR_DTB_FILENAME;
+	if (!load_file(dtb_filename, PARAMS_ADDR)) {
+		Debug("Failed to load %s", dtb_filename);
+		goto error;
 	}
 
-	if (!load_file(FALLBACK_PATH ARM9LINUXFW_FILENAME, ARM9LINUXFW_ADDR)) {
-		if (!load_file(ARM9LINUXFW_FILENAME, ARM9LINUXFW_ADDR))
-			Debug("Continuing without an arm9linuxfw...");
+	if (!load_file(ARM9LINUXFW_FILENAME, ARM9LINUXFW_ADDR)) {
+		Debug("Continuing without an arm9linuxfw...");
```

Comment:
is there any reason to not make arm9linuxfw mandatory at this point?

Reply:
not really, we can force it to be needed now that it's fairly stable

Reply:
SGTM: filed: https://github.com/linux-3ds/firm_linux_loader/issues/8